pkg/reconciler/openshift/tektonpipeline/extension.go

```diff
@@ -168,32 +179,27 @@ func (oe openshiftExtension) PostReconcile(ctx context.Context, comp v1alpha1.Te
 			}
 
 			if err := createInstallerSet(ctx, oe.operatorClientSet, pipeline, oe.manifest, oe.version,
-				postPipelineInstallerSet, "post-pipeline"); err != nil {
+				postPipelineInstallerSet); err != nil {
 				return err
 			}
 		}
 
 	} else {
-		return deleteInstallerSet(ctx, oe.operatorClientSet, pipeline, postPipelineInstallerSet)
+		return deleteInstallerSet(ctx, oe.operatorClientSet, pipeline, postPipelineInstallerSet, postReconcileSelector)
 	}
 
 	return nil
 }
 func (oe openshiftExtension) Finalize(ctx context.Context, comp v1alpha1.TektonComponent) error {
 	pipeline := comp.(*v1alpha1.TektonPipeline)
-
-	installerSets := pipeline.Status.ExtentionInstallerSets
-	if len(installerSets) == 0 {
-		return nil
+	err := deleteInstallerSet(ctx, oe.operatorClientSet, pipeline, postPipelineInstallerSet, postReconcileSelector)
+	if err != nil {
+		return err
 	}
```

Comment:
I think we need `prePipelineInstallerSet` & `preReconcileSelector` here as post thing are handled below

```suggestion
	if err := deleteInstallerSet(ctx, oe.operatorClientSet, pipeline, prePipelineInstallerSet, preReconcileSelector); err != nil {
		return err
	}
```